packages/react/src/primitives/Select/Select.tsx

```diff
@@ -35,12 +35,16 @@ const SelectPrimitive: Primitive<SelectProps, 'select'> = (
   // value === undefined is to make sure that component is used in uncontrolled way so that setting defaultValue is valid
   const shouldSetDefaultPlaceholderValue =
     value === undefined && defaultValue === undefined && placeholder;
+
+  const isExpanded = isMultiple || selectSize > 1;
+
   const componentClasses = classNames(
     ComponentClassNames.Select,
     ComponentClassNames.FieldGroupControl,
     classNameModifier(ComponentClassNames.Select, size),
     classNameModifier(ComponentClassNames.Select, variation),
     classNameModifierByFlag(ComponentClassNames.Select, 'error', hasError),
+    classNameModifierByFlag(ComponentClassNames.Select, 'expanded', isExpanded),
```

Comment:
We were styling the "expanded" view using attribute selectors (`.amplify-selector[size]`, `.amplify-selector[multiple]`) but we don't want just any [size] attribute, we want it when it is greater than 1. Since we can't do those comparisons in CSS we'll use the `isExpanded` flag we defined earlier to create a class.